hail/src/main/scala/is/hail/expr/ir/ComparisonOp.scala

```diff
@@ -89,4 +92,36 @@ case class Compare(t1: Type, t2: Type) extends ComparisonOp[Int] {
   override val strict: Boolean = false
   val op: CodeOrdering.Op = CodeOrdering.Compare()
 }
-object Compare { def apply(typ: Type): Compare = Compare(typ, typ) }
\ No newline at end of file
+object Compare { def apply(typ: Type): Compare = Compare(typ, typ) }
+
+trait StructComparisonOp[T] extends ComparisonOp[T] {
+  val sortFields: Array[SortField]
+
+  override def codeOrdering(ecb: EmitClassBuilder[_], t1: SType, t2: SType): F[op.ReturnType] = {
+    ComparisonOp.checkCompatible(t1.virtualType, t2.virtualType)
+    ecb.getStructOrderingFunction(t1.asInstanceOf[SBaseStruct], t2.asInstanceOf[SBaseStruct], sortFields, op).asInstanceOf[CodeOrdering.F[op.ReturnType]]
+  }
+}
+
+case class StructCompare(t1: Type, t2: Type, sortFields: Array[SortField]) extends StructComparisonOp[Int] {
+  val op: CodeOrdering.Op = CodeOrdering.StructCompare()
+  override val strict: Boolean = false
```

Comment:
Admittedly am not 100% sure what `strict` means in this context.

Reply:
strict refers to missing semantics -- strict=true means the result is missing if any argument is missing.

Reply:
I think this all looks correct.